lib/ecto/migrator.ex

```diff
@@ -138,13 +138,15 @@ defmodule Ecto.Migrator do
 
     case repo.start_link(pool_size: pool_size) do
       {:ok, _} ->
+        ensure_migration_repo_started(repo, repo.config[:migration_repo])
```

Comment:
You should be able to eliminate a clause for `ensure_migration_repo_started/2` if you do this:

```suggestion
        ensure_migration_repo_started(repo, repo.config[:migration_repo] || repo)
```